### PR TITLE
Simplify the DCA handling for return nodes.

### PR DESCRIPTION
**Pull request by tritao**
## Description

This simplifies the handling of return nodes during dead code analysis.

Here is the updated graphs for this example:

```rust
script;
fn main() -> u64 {
    return 1;
    return 0
}
```

Before:

test

After:

test2

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Comment by ironcev** [comment]
# TL;DR
- The PR removes the double edge from the `return` node to its contained nodes which was a bug :+1: 
- Moving the "return" edge from the `return` to the content might actually lead to more complex graphs if we have more then one node in the content.
- Can we also remove the "return" edge going from the implicit return to the `"main" fn exit`?
- I see PR as going in the right direction, because to properly handle divergence we will need to dig into the `return`'s content and not only directly connect `return` to the exit node.

# Detailed thoughts
Ha ha, I was thinking about that very same TODO comment today, while pondering on the background thread about our `deterministically_aborts` + `Never` + `match` + `TypeInfo::Unknown` combo of issues :smile: Though my thinking was more on the question of divergence which was not in focus of this PR.

My thoughts. Removing the double edge from the `return` node to its content is a simplification. That was definitely a bug before, adding an edge after it was already added in the `connect_expression`.

Moving the "return" line from the `return` to the content node might be seen as a simplification, but it looks to me more as a matter of taste. For example, if there are more than one node in the `return` content like in this example, one might argue that we actually got a more complex graph with more edges (assuming we remove the double edge):

```Sway
fn main() -> [u64;3] {
    return [1, 2, 3];
    return [1, 2, 3]
}
``` 
Before:
DCA graph before

After:
DCA graph after

But still I think the new approach moves us in the right direction, where we will need to inspect the `return` content and check for divergence. This is what I planed to do after sorting out the `deterministically_abort` issue.

What I want to say, in an example like the one below, we cannot simply connect all the parts of the content to the exit node or mark them as reachable. The variable `a` is not reachable and it should be reported as such, but DCA currently does not marks it as such, because in both the new and old implementation of `return` it assumes that whatever is in `return` will be reachable which must not be the case.

```Sway
fn main() -> [u64;3] {
    let a = 123u64;
    return [0u64, return [1, 2, 3], a];
}
```
For the same example, Rust will properly report that `a` is not reachable because of the preceeding `return [1, 2, 3]`.

But sorry, I went a bit too far, divergence is way out of scope of this PR :smile: 

A detail perhaps being relevant to this PR is that one "return" edge going from the implicit `return` to the `"main" fn exit`. Can we also remove that edge or it needs to stay?

**Review by ironcev — APPROVED** [review]
Approved as per the comment. Removing the direct "return" edge between the implicit `return` and the exit might be worth thinking of.
